### PR TITLE
[BSE-4371] Replace Secret File with GitHub Secrets

### PR DESCRIPTION
**Pull request by srilman**
## Changes included in this PR

Rather than using a single secret file with multiple values that we parse out, we should just use GitHub secrets in general

## Testing strategy

- [x] Running Conda Build Release CI to Success (I also tested if the secrets are read in the scripts).

## User facing changes

Internal only.

## Checklist
- [x] Pipelines passed before requesting review. To run CI you must include `[run CI]` in your commit message.
- [x] I am familiar with the [Contributing Guide](https://github.com/bodo-ai/Bodo/blob/main/CONTRIBUTING.md) 
- [x] I have installed + ran pre-commit hooks.

### CODE REVIEW

File: .github/workflows/_build_azurefs_sas_token_provider_conda.yml [hunk 1]
```diff
@@ -34,10 +31,12 @@ jobs:
       - name: 'Set Secret File Permissions and Conda Build and Publish Bodo-AzureFS-SAS-Token-Provider Binary to Artifactory'
         run: |
           set -eo pipefail
-          echo "${{ secrets.PUBLISH_BINARY_SECRETS }}" > $HOME/secret_file
-          sudo chmod a+r $HOME/secret_file
 
           artifactory_channel="bodo.ai"
-          echo "artifactory_channel: $artifactory_channel"
-          ./buildscripts/azurefs-sas-token-provider/publish_binary.sh $artifactory_channel
+          ./buildscripts/azurefs-sas-token-provider/publish_binary.sh "bodo.ai"
```
Comment (hadia206) [inline .github/workflows/_build_azurefs_sas_token_provider_conda.yml:36]: Shouldn't this use `BODO_CHANNEL_NAME`
```suggestion
          ./buildscripts/azurefs-sas-token-provider/publish_binary.sh $BODO_CHANNEL_NAME
```

**Review by hadia206 — APPROVED** [review]
Looks good to me.
Thanks a lot

**Review by IsaacWarren — APPROVED** [review]
Thanks Srinivas

File: .github/workflows/_build_azurefs_sas_token_provider_conda.yml (hunk 1 above)
Comment (srilman) [reply .github/workflows/_build_azurefs_sas_token_provider_conda.yml:36]: I don't think it was before

**Pull request merged**